src/main/webapp/scripts/controller-auth.js

```diff
@@ -0,0 +1,46 @@
+import { postVideoComment } from '/scripts/post-comment.js';
+import { videoId, commentElement } from '/scripts/view.js';
+import { successfulApiCallback, failedApiCallback, updateSigninStatus } from '/scripts/view.js';
+
+const apiKey = '';
+const clientId = '';
+const discoveryDocs = ["https://www.googleapis.com/discovery/v1/apis/youtube/v3/rest"];
+const scopes = "https://www.googleapis.com/auth/youtube.force-ssl";
+
+function handleClientLoad() {
+  gapi.load('client:auth2', initClient);
+}
+
+function initClient() {
+  gapi.client.init({
+    apiKey: apiKey,
+    discoveryDocs: discoveryDocs,
+    clientId: clientId,
+    scope: scopes
+  })
+    .then(function () {
+      // Listen for sign-in state changes.
+      gapi.auth2.getAuthInstance().isSignedIn.listen(updateSigninStatus);
+
+      // Handle the initial sign-in state.
+      updateSigninStatus(gapi.auth2.getAuthInstance().isSignedIn.get(),
+        handleAuthClick, handleSignoutClick, handleCommentClick);
+    });
+}
+
+function handleAuthClick(event) {
+  gapi.auth2.getAuthInstance().signIn();
+}
+
+function handleSignoutClick(event) {
+  gapi.auth2.getAuthInstance().signOut();
+}
+
+function handleCommentClick(event) {
+  postVideoComment(videoId, commentElement.value)
+    .then(data => successfulApiCallback(data))
```

Comment:
Because `successfulApiCallback` takes the same argument as the promise result, you can simplify this line to `then(successFulApiCallBack)`

Reply:
Done. This looks much nicer